com.hazelcast.mapstore.* 'NoClassDefFoundError' [HZ-3660]

_master_ (commit 0d36189619cd30230e4a68818658c2183c427337)
Failed on oracle-11: https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-16/2/testReport/junit/com.hazelcast.mapstore.postgres/

<details><summary>Example Stacktrace:</summary>

```
java.lang.NoClassDefFoundError: com/hazelcast/shaded/org/apache/calcite/rex/RexVisitor
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3137)
	at java.base/java.lang.Class.getConstructor0(Class.java:3342)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2553)
	at com.hazelcast.internal.util.ServiceLoader$NewInstanceIterator.next(ServiceLoader.java:293)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at com.hazelcast.jet.sql.impl.connector.SqlConnectorCache.<init>(SqlConnectorCache.java:38)
	at com.hazelcast.jet.sql.impl.CalciteSqlOptimizer.<init>(CalciteSqlOptimizer.java:248)
	at com.hazelcast.sql.impl.SqlServiceImpl.start(SqlServiceImpl.java:95)
	at com.hazelcast.spi.impl.NodeEngineImpl.start(NodeEngineImpl.java:292)
	at com.hazelcast.instance.impl.Node.start(Node.java:483)
	at com.hazelcast.instance.impl.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:125)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:218)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:197)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:236)
	at com.hazelcast.jet.SimpleTestInClusterSupport.initializeWitSupplier(SimpleTestInClusterSupport.java:80)
	at com.hazelcast.jet.SimpleTestInClusterSupport.initialize(SimpleTestInClusterSupport.java:68)
	at com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport.initialize(JdbcSqlTestSupport.java:78)
	at com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport.initialize(JdbcSqlTestSupport.java:65)
	at com.hazelcast.mapstore.postgres.PostgresGenericMapStoreTest.beforeClass(PostgresGenericMapStoreTest.java:30)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:41)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: com.hazelcast.shaded.org.apache.calcite.rex.RexVisitor
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 25 more

```

</details>

<details><summary>Standard output:</summary>

```
11:03:58,970  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.4.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
11:03:58,972  INFO || - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Hazelcast Platform 5.4.0-SNAPSHOT (20231102 - 0d36189) starting at [127.0.0.1]:5701
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Cluster name: dev
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
11:03:58,973  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Jet is enabled
11:03:58,983  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
11:03:58,984  INFO || - [TpcServerBootstrap] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] TPC: disabled
11:03:58,998  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
11:03:59,019  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
11:03:59,055  WARN || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Terminating forcefully...
11:03:59,055  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Shutting down connection manager...
11:03:59,055  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Shutting down node engine...
11:03:59,061  INFO || - [NodeExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Destroying node NodeExtension.
11:03:59,061  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Hazelcast Shutdown is completed in 6 ms.
11:03:59,577  INFO || - [SimpleTestInClusterSupport] Time-limited test - Terminating instance factory in SimpleTestInClusterSupport.@AfterClass

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-postgres-16/2/

[com.hazelcast.mapstore.postgres.PostgreGenericMapStoreIntegrationTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-16/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.postgres/PostgreGenericMapStoreIntegrationTest/___/)
[com.hazelcast.mapstore.postgres.PostgresGenericMapStoreTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-16/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.postgres/PostgresGenericMapStoreTest/___/)
[com.hazelcast.mapstore.mysql.MySQLGenericMapStoreIntegrationTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-mysql-8.2/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.mysql/MySQLGenericMapStoreIntegrationTest/___/)
[com.hazelcast.mapstore.mysql.MySQLGenericMapStoreTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-mysql-8.2/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.mysql/MySQLGenericMapStoreTest/___/)

Reopening as issue occurred again on: _master_ (commit 1dfbd0992f66dbf8747aee16dfead6d953db4d9d)
with oracle-11: https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-14/30/testReport/com.hazelcast.mapstore.postgres/PostgreGenericMapStoreIntegrationTest/___/
<details><summary>Stacktrace:</summary>

```
java.lang.NoClassDefFoundError: com/hazelcast/shaded/org/apache/calcite/rex/RexVisitor
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3137)
	at java.base/java.lang.Class.getConstructor0(Class.java:3342)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2553)
	at com.hazelcast.internal.util.ServiceLoader$NewInstanceIterator.next(ServiceLoader.java:293)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at com.hazelcast.jet.sql.impl.connector.SqlConnectorCache.<init>(SqlConnectorCache.java:38)
	at com.hazelcast.jet.sql.impl.CalciteSqlOptimizer.<init>(CalciteSqlOptimizer.java:250)
	at com.hazelcast.sql.impl.SqlServiceImpl.start(SqlServiceImpl.java:95)
	at com.hazelcast.spi.impl.NodeEngineImpl.start(NodeEngineImpl.java:292)
	at com.hazelcast.instance.impl.Node.start(Node.java:483)
	at com.hazelcast.instance.impl.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:124)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:218)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:197)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:236)
	at com.hazelcast.jet.SimpleTestInClusterSupport.initializeWitSupplier(SimpleTestInClusterSupport.java:80)
	at com.hazelcast.jet.SimpleTestInClusterSupport.initializeWithClientAndConfigSupplier(SimpleTestInClusterSupport.java:99)
	at com.hazelcast.jet.SimpleTestInClusterSupport.initializeWithClient(SimpleTestInClusterSupport.java:90)
	at com.hazelcast.mapstore.GenericMapStoreIntegrationTest.initializeBeforeClass(GenericMapStoreIntegrationTest.java:106)
	at com.hazelcast.mapstore.postgres.PostgreGenericMapStoreIntegrationTest.beforeClass(PostgreGenericMapStoreIntegrationTest.java:35)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:41)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: com.hazelcast.shaded.org.apache.calcite.rex.RexVisitor
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 25 more

```
</details>


<details><summary>Standard output:</summary>

```
13:53:50,017  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.4.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
13:53:50,066  INFO || - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
13:53:50,066  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
13:53:50,080  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Hazelcast Platform 5.4.0-SNAPSHOT (20231121 - 1dfbd09) starting at [127.0.0.1]:5701
13:53:50,080  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Cluster name: dev
13:53:50,080  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
13:53:50,081  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Jet is enabled
13:53:50,099  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
13:53:50,116  INFO || - [TpcServerBootstrap] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] TPC: disabled
13:53:50,358  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
13:53:50,635  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 8
13:53:50,674  WARN || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Terminating forcefully...
13:53:50,676  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Shutting down connection manager...
13:53:50,676  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Shutting down node engine...
13:53:50,688  INFO || - [NodeExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Destroying node NodeExtension.
13:53:50,688  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Hazelcast Shutdown is completed in 14 ms.
13:53:51,339  INFO || - [SimpleTestInClusterSupport] Time-limited test - Terminating instance factory in SimpleTestInClusterSupport.@AfterClass

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-postgres-14/30/
^ Build was triggered 20m ago, so for sure after merged fix.

--- a/extensions/mapstore/pom.xml
+++ b/extensions/mapstore/pom.xml
@@ -51,27 +51,31 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <configuration>
-                    <!--
-                    hazelcast-mapstore doesn't have any dependencies which should be included,
-                    no need for jar-with-dependencies
-                    -->
-                    <shadedArtifactAttached>false</shadedArtifactAttached>
-                    <relocations>
-                        <relocation>
+                <executions>
+                    <execution>
+                        <id>jar-with-dependencies</id>
+                        <phase>${shade.jar-with-dependencies.phase}</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration combine.self="override">
                             <!--
-                            We need to relocate our usage of org.apache.calcite to the location used in hazelcast-sql
+                            hazelcast-mapstore doesn't have any dependencies which should be included,
+                            no need for jar-with-dependencies
                             -->
-                            <pattern>org.apache.calcite</pattern>
-                            <shadedPattern>${relocation.root}.org.apache.calcite</shadedPattern>
-                        </relocation>
-                    </relocations>
-                </configuration>
-            </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-failsafe-plugin</artifactId>
-                <version>${maven.failsafe.plugin.version}</version>
+                            <shadedArtifactAttached>false</shadedArtifactAttached>
+                            <relocations>
+                                <relocation>
+                                    <!--
+                                    We need to relocate our usage of org.apache.calcite to the location used in hazelcast-sql
+                                    -->
+                                    <pattern>org.apache.calcite</pattern>
+                                    <shadedPattern>${relocation.root}.org.apache.calcite</shadedPattern>
+                                </relocation>
+                            </relocations>
+                        </configuration>
+                    </execution>
+                </executions>
             </plugin>
         </plugins>
     </build>

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreIT.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreIT.java
@@ -65,7 +65,7 @@ import static org.assertj.core.util.Lists.newArrayList;
 
 @RunWith(HazelcastSerialClassRunner.class)
 @Category({QuickTest.class})
-public class GenericMapStoreIntegrationTest extends JdbcSqlTestSupport {
+public class GenericMapStoreIT extends JdbcSqlTestSupport {
 
     private static Config memberConfig;
 

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/mariadb/MariaDBGenericMapStoreIT.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/mariadb/MariaDBGenericMapStoreIT.java
@@ -16,16 +16,16 @@
 
 package com.hazelcast.mapstore.mariadb;
 
-import com.hazelcast.mapstore.GenericMapStoreIntegrationTest;
+import com.hazelcast.mapstore.GenericMapStoreIT;
 import com.hazelcast.test.annotation.NightlyTest;
 import com.hazelcast.test.jdbc.MariaDBDatabaseProvider;
 import org.junit.BeforeClass;
 import org.junit.experimental.categories.Category;
 
 @Category({NightlyTest.class})
-public class MariaDBGenericMapStoreIntegrationTest extends GenericMapStoreIntegrationTest {
+public class MariaDBGenericMapStoreIT extends GenericMapStoreIT {
 
-    public MariaDBGenericMapStoreIntegrationTest() {
+    public MariaDBGenericMapStoreIT() {
         setPrefix("maria_");
     }
 

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/mssql/MSSQLGenericMapStoreIT.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/mssql/MSSQLGenericMapStoreIT.java
@@ -14,24 +14,26 @@
  * limitations under the License.
  */
 
-package com.hazelcast.mapstore.postgres;
+package com.hazelcast.mapstore.mssql;
 
-import com.hazelcast.mapstore.GenericMapStoreIntegrationTest;
+import com.hazelcast.mapstore.GenericMapStoreIT;
 import com.hazelcast.test.annotation.NightlyTest;
-import com.hazelcast.test.jdbc.PostgresDatabaseProvider;
+import com.hazelcast.test.jdbc.MSSQLDatabaseProvider;
 import org.junit.BeforeClass;
+import org.junit.Ignore;
 import org.junit.experimental.categories.Category;
 
+@Ignore
 @Category({NightlyTest.class})
-public class PostgreGenericMapStoreIntegrationTest extends GenericMapStoreIntegrationTest {
+public class MSSQLGenericMapStoreIT extends GenericMapStoreIT {
 
-    public PostgreGenericMapStoreIntegrationTest() {
-        setPrefix("postgre_");
+    public MSSQLGenericMapStoreIT() {
+        setPrefix("mssql_");
     }
 
     // Shadow the parent's @BeforeClass method by using the same method name
     @BeforeClass
     public static void beforeClass() {
-        initializeBeforeClass(new PostgresDatabaseProvider());
+        initializeBeforeClass(new MSSQLDatabaseProvider());
     }
 }

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/mysql/MySQLGenericMapStoreIT.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/mysql/MySQLGenericMapStoreIT.java
@@ -14,20 +14,24 @@
  * limitations under the License.
  */
 
-package com.hazelcast.mapstore.postgres;
+package com.hazelcast.mapstore.mysql;
 
-import com.hazelcast.mapstore.GenericMapStoreTest;
+import com.hazelcast.mapstore.GenericMapStoreIT;
 import com.hazelcast.test.annotation.NightlyTest;
-import com.hazelcast.test.jdbc.PostgresDatabaseProvider;
+import com.hazelcast.test.jdbc.MySQLDatabaseProvider;
 import org.junit.BeforeClass;
 import org.junit.experimental.categories.Category;
 
-@Category(NightlyTest.class)
-public class PostgresGenericMapStoreTest extends GenericMapStoreTest {
+@Category({NightlyTest.class})
+public class MySQLGenericMapStoreIT extends GenericMapStoreIT {
 
-    @BeforeClass
-    public static void beforeClass()  {
-        initialize(new PostgresDatabaseProvider());
+    public MySQLGenericMapStoreIT() {
+        setPrefix("mysql_");
     }
 
+    // Shadow the parent's @BeforeClass method by using the same method name
+    @BeforeClass
+    public static void beforeClass() {
+        initializeBeforeClass(new MySQLDatabaseProvider());
+    }
 }

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/postgres/PostgreGenericMapStoreIT.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/postgres/PostgreGenericMapStoreIT.java
@@ -14,24 +14,24 @@
  * limitations under the License.
  */
 
-package com.hazelcast.mapstore.mysql;
+package com.hazelcast.mapstore.postgres;
 
-import com.hazelcast.mapstore.GenericMapStoreIntegrationTest;
+import com.hazelcast.mapstore.GenericMapStoreIT;
 import com.hazelcast.test.annotation.NightlyTest;
-import com.hazelcast.test.jdbc.MySQLDatabaseProvider;
+import com.hazelcast.test.jdbc.PostgresDatabaseProvider;
 import org.junit.BeforeClass;
 import org.junit.experimental.categories.Category;
 
 @Category({NightlyTest.class})
-public class MySQLGenericMapStoreIntegrationTest extends GenericMapStoreIntegrationTest {
+public class PostgreGenericMapStoreIT extends GenericMapStoreIT {
 
-    public MySQLGenericMapStoreIntegrationTest() {
-        setPrefix("mysql_");
+    public PostgreGenericMapStoreIT() {
+        setPrefix("postgre_");
     }
 
     // Shadow the parent's @BeforeClass method by using the same method name
     @BeforeClass
     public static void beforeClass() {
-        initializeBeforeClass(new MySQLDatabaseProvider());
+        initializeBeforeClass(new PostgresDatabaseProvider());
     }
 }

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/postgres/PostgresGenericMapStoreSimpleIT.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/postgres/PostgresGenericMapStoreSimpleIT.java
@@ -14,26 +14,20 @@
  * limitations under the License.
  */
 
-package com.hazelcast.mapstore.mssql;
+package com.hazelcast.mapstore.postgres;
 
-import com.hazelcast.mapstore.GenericMapStoreIntegrationTest;
+import com.hazelcast.mapstore.GenericMapStoreTest;
 import com.hazelcast.test.annotation.NightlyTest;
-import com.hazelcast.test.jdbc.MSSQLDatabaseProvider;
+import com.hazelcast.test.jdbc.PostgresDatabaseProvider;
 import org.junit.BeforeClass;
-import org.junit.Ignore;
 import org.junit.experimental.categories.Category;
 
-@Ignore
-@Category({NightlyTest.class})
-public class MSSQLGenericMapStoreIntegrationTest extends GenericMapStoreIntegrationTest {
+@Category(NightlyTest.class)
+public class PostgresGenericMapStoreSimpleIT extends GenericMapStoreTest {
 
-    public MSSQLGenericMapStoreIntegrationTest() {
-        setPrefix("mssql_");
-    }
-
-    // Shadow the parent's @BeforeClass method by using the same method name
     @BeforeClass
-    public static void beforeClass() {
-        initializeBeforeClass(new MSSQLDatabaseProvider());
+    public static void beforeClass()  {
+        initialize(new PostgresDatabaseProvider());
     }
+
 }

--- a/extensions/mongodb/pom.xml
+++ b/extensions/mongodb/pom.xml
@@ -45,6 +45,7 @@
                             <Automatic-Module-Name>com.hazelcast.jet.mongodb</Automatic-Module-Name>
                         </manifestEntries>
                     </archive>
+                    <forceCreation>true</forceCreation>
                 </configuration>
             </plugin>
 

--- a/pom.xml
+++ b/pom.xml
@@ -1320,7 +1320,7 @@
                                 com.hazelcast.test.annotation.SlowTest
                             </groups>
                             <includes>
-                                <include>**/**.java</include>
+                                <include>**/**Test.java</include>
                             </includes>
                             <excludes>
                                 <exclude>**/**IT.java</exclude>

--- a/pom.xml
+++ b/pom.xml
@@ -1346,7 +1346,7 @@
                             <useManifestOnlyJar>false</useManifestOnlyJar>
                             <useSystemClassLoader>true</useSystemClassLoader>
                             <includes>
-                                <include>**/**IT.java</include>
+                                <include>**/*IT.java</include>
                             </includes>
                             <groups>
                                 com.hazelcast.test.annotation.NightlyTest,

--- a/pom.xml
+++ b/pom.xml
@@ -1320,7 +1320,7 @@
                                 com.hazelcast.test.annotation.SlowTest
                             </groups>
                             <includes>
-                                <include>**/**Test.java</include>
+                                <include>**/**.java</include>
                             </includes>
                             <excludes>
                                 <exclude>**/**IT.java</exclude>

--- a/pom.xml
+++ b/pom.xml
@@ -1315,17 +1315,17 @@
                                 -Dlog4j.skipJansi=true
                                 ${extraVmArgs}
                             </argLine>
+                            <groups>
+                                com.hazelcast.test.annotation.NightlyTest,
+                                com.hazelcast.test.annotation.SlowTest
+                            </groups>
                             <includes>
-                                <include>**/**.java</include>
+                                <include>**/**Test.java</include>
                             </includes>
                             <excludes>
                                 <exclude>**/**IT.java</exclude>
                                 <exclude>**/jsr/**.java</exclude>
                             </excludes>
-                            <groups>
-                                com.hazelcast.test.annotation.NightlyTest,
-                                com.hazelcast.test.annotation.SlowTest
-                            </groups>
                         </configuration>
                     </plugin>
                     <plugin>
@@ -1345,6 +1345,9 @@
                             </argLine>
                             <useManifestOnlyJar>false</useManifestOnlyJar>
                             <useSystemClassLoader>true</useSystemClassLoader>
+                            <includes>
+                                <include>**/**IT.java</include>
+                            </includes>
                             <groups>
                                 com.hazelcast.test.annotation.NightlyTest,
                                 com.hazelcast.test.annotation.SlowTest

